app/Libraries/ApidocRouteHelper.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+namespace App\Libraries;
+
+use App\Http\Middleware\RequireScopes;
+
+class ApidocRouteHelper
+{
+    private $routeScopes = [];
+
+    public static function instance()
+    {
+        static $instance;
+
+        if ($instance === null) {
+            $instance = new static();
+        }
+
+        return $instance;
+    }
+
+    private function __construct()
+    {
+        $routeScopesHelper = new RouteScopesHelper();
+        $routeScopesHelper->loadRoutes();
+
+        foreach ($routeScopesHelper->toArray() as $route) {
+            $key = $this->keyFor($route['uri'], $route['method']);
+            $route['scopes'] = array_filter($route['scopes'], function ($scope) {
+                return $scope !== 'any';
+            });
+            $this->routeScopes[$key] = $route;
+        }
+    }
+
+    public function getScopes(string $uri, string $method)
```

Comment:
maybe add "lazer only" label when empty? Will need to differentiate with `any` though.